bin/varnishd/cache/cache_backend.c

```diff
@@ -229,23 +229,31 @@ vbe_dir_getfd(VRT_CTX, struct worker *wrk, VCL_BACKEND dir, struct backend *bp,
 	if (!VTAILQ_EMPTY(&bp->cw_head) || BE_BUSY(bp))
 		cw->cw_state = CW_BE_BUSY;
 
-	if (cw->cw_state == CW_BE_BUSY && wait_limit > 0 &&
-	    wait_tmod > 0.0 && bp->cw_count < wait_limit) {
-		VTAILQ_INSERT_TAIL(&bp->cw_head, cw, cw_list);
-		bp->cw_count++;
-		VSC_C_main->backend_wait++;
-		cw->cw_state = CW_QUEUED;
-		wait_end = VTIM_real() + wait_tmod;
-		do {
-			err = Lck_CondWaitUntil(&cw->cw_cond, bp->director->mtx,
-			    wait_end);
-		} while (err == EINTR);
-		bp->cw_count--;
-		if (err != 0 && BE_BUSY(bp)) {
-			VTAILQ_REMOVE(&bp->cw_head, cw, cw_list);
-			VSC_C_main->backend_wait_fail++;
-			cw->cw_state = CW_BE_BUSY;
+	switch (cw->cw_state) {
```

Comment:
Actually I disagree because it's not just a switch/case, but rather additional conditions in the if clause

Reply:
Please have a look at the finished product: https://github.com/varnishcache/varnish-cache/blob/6808a19dfa5f90de4781bceffb85918b870d9712/bin/varnishd/cache/cache_backend.c#L182-L213
Given the name indicate it is a state machine, I think a switch suits better for this purpose.